src/sentry/rules/processing/delayed_processing.py

```diff
@@ -177,6 +192,41 @@ def get_rules_to_fire(
     return rules_to_fire
 
 
+def get_group_to_groupevent(
+    rulegroup_to_events: dict[str, str], project_id: int, group_ids: set[int]
+) -> dict[Group, GroupEvent]:
+    group_to_groupevent: dict[Group, GroupEvent] = {}
+    groups = Group.objects.filter(id__in=group_ids)
+    group_id_to_group = {group.id: group for group in groups}
+    for rule_group, instance_id in rulegroup_to_events.items():
```

Comment:
Does it make sense to parse these in the function that fetches `rulegroup_to_events`, and have the type be
`dict[(tuple(int, int): <json>]`. That way you don't need to parse it all in here

Reply:
Do you mean in `get_rules_to_groups`? There isn't a function besides `buffer.get_hash` that gets `rulegroup_to_events`.

Reply:
Ahh I hadn't checked and assumed there was a function for this. It's nbd, but it might be better to transform it separately and just pass it in. Feel free to ignore though